komodo-core/src/main/java/org/komodo/repository/RepositoryImpl.java

```diff
@@ -570,6 +587,36 @@ public RepositoryImpl( final Type type,
         this.id = id;
     }
 
+    protected void checkSecurity(UnitOfWork transaction, String nodePath) throws KException {
+        String userWksp = komodoWorkspacePath(transaction);
+
+        if (isSystemTx(transaction))
+            return; // System can do what it wishes
+
+        if (! nodePath.startsWith(WORKSPACE_ROOT))
+            return; // Adding node somewhere other than workspace
```

Comment:
Wouldn't this allow the user to delete and modify Teiid instances and library objects (without checkout) directly? Do we want library objects to be "owned" by all users to do as they please? And couldn't they also add objects into those directories? Agree this hasn't really been thought out but I tend to think we should lean toward restricting access and add in later as we see fit.